src/shogun/classifier/NearestCentroid.cpp

```diff
@@ -72,18 +72,16 @@ namespace shogun{
 			num_per_class[current_class]++;
 		}
 
+		SGVector<float64_t> scale(num_classes);
 		for (int32_t i=0 ; i<num_classes ; i++)
 		{
-			auto target = centroids.get_column(i);
 			int32_t total = num_per_class[i];
-			float64_t scale = 0;
 			if(total>1)
-				scale = 1.0/((float64_t)(total-1));
+				scale[i] = 1.0/((float64_t)(total-1));
 			else
-				scale = 1.0/(float64_t)total;
-
-			target.scale(scale);
+				scale[i] = 1.0/(float64_t)total;
 		}
+		linalg::scale(centroids, centroids, scale);
```

Comment:
sweet!

Reply:
i think your signature doesnt match the one in linalg where result is the last ?

Reply:
the signature of `scale` in `LinalgBackendBase` has the result at the end, while the one in `linalg` namespace has the scaling factor in the end.